xfce.thunar-vcs-plugin: init at 0.3.0

LordMZTE opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
This adds [thunar-vcs-plugin](https://gitlab.xfce.org/thunar-plugins/thunar-vcs-plugin/) to the `xfce` package group. The plugin adds functionality for Git and Subversion to Thunar.

Closes #334093

A review by the Xfce team would be appreciated!
@bobby285271 @romildo @muscaln

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

LordMZTE committed:
xfce.thunar-vcs-plugin: init at 0.3.0

bobby285271 commented:
Thanks for the PR! A few suggestions:

- There is no need to run `./autogen.sh`, since `xfce4-dev-tools` contains a [setup-hook](https://github.com/NixOS/nixpkgs/blob/f170c23e25d49aa5bc26b70cb3058ad461668490/pkgs/desktops/xfce/core/xfce4-dev-tools/setup-hook.sh) that already triggers `xdt-autogen`.
- There are various [optional dependencies](https://github.com/xfce-mirror/thunar-vcs-plugin/blob/6253dcb2166b7abae725080408ff2e8d1103e1c5/configure.ac#L84) that are not included here, I checked some of the distro packaging [1](https://packages.debian.org/sid/source/thunar-vcs-plugin) [2](https://src.fedoraproject.org/rpms/thunar-vcs-plugin/blob/rawhide/f/thunar-vcs-plugin.spec) and I guess having them enabled can probably be fine.

I will try to test this later.

bobby285271 commented:
> I will try to test this later.

I quickly tried this and I think I hit e.g. [thunar-vcs-plugin#11](https://gitlab.xfce.org/thunar-plugins/thunar-vcs-plugin/-/issues/11), seeing empty window in various places. It is unclear to me whether this is expected (since the issue is quite old). I won't block on this (since the project has release 1 month ago) but just to make sure are you happy with your testing of the plugin?

bobby285271 committed:
xfce.thunar-vcs-plugin: Explicitly add gtk3, glib

These are direct dependencies so explicitly add them instead of
relying on propagation.

LordMZTE commented:
Indeed the plugin seems pretty unfinished - but some things do work like the log view. I doubt any of that is caused by improper packaging, though.

bobby285271 commented:
Let's merge then, thanks!
